client/core.lisp

```diff
@@ -25,27 +25,57 @@
 
 
 (eval-when (:compile-toplevel :load-toplevel :execute)
-  (defun generate-method-description (spec)
-    (let* ((original-name (gethash "name" spec))
-           (name (intern (normalize-name original-name)))
-           (params-spec (gethash "params" spec))
-           (lambda-list (generate-lambda-list params-spec)))
-      `(format stream "- ~S~%"
-               ',(cons name
-                       lambda-list))))
+  (declaim (ftype (function (closer-mop:specializer stream) null)
+		  generate-method-descriptions))
+  (defun generate-method-descriptions (class stream)
```

Comment:
You don't need to wrap `generate-method-description` with `eval-when` anymore, because this function is not called during the macro-expansion.